src/fontra/views/editor/editor.js

```diff
@@ -421,17 +421,36 @@ export class EditorController {
       this.updateWindowLocationAndSelectionInfo();
       this.autoViewBox = false;
     });
-    this.sceneController.addEventListener("createNewSource", (event) => {
-      this.sidebarDesignspace.addSource();
-    });
-    this.sceneController.addEventListener("goToNearestSource", async (event) => {
-      const glyphController =
-        await this.sceneController.sceneModel.getSelectedVariableGlyphController();
-      const nearestSourceIndex = glyphController.findNearestSourceFromGlobalLocation(
-        this.sceneController.getLocation(),
-        true
+    this.sceneController.addEventListener("cantEditGlyphNotAtSource", async () => {
```

Comment:
Please move the addEventHandler call to right after the addEventHandler call for "doubleClickedComponents", and move the handler implementation to a new method called "handleCantEditGlyphNotAtSource()" (or something else if you can come up with a better name).

Reply:
Ok I did. I created "showGlyphHasNoSourceDialog" method and I added "glyphHasNoSource" event.

Reply:
Thanks!

I don't like the name though. How about "glyphEditLocationNotAtSource" + "showGlyphEditLocationNotAtSource". It is not the glyph that doesn't have a source, but that there is no source at the desired edit location.

Reply:
Sounds better.